gcp/deploy.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
```

Comment:
Where is this used?

Reply:
It's used in the next PR that adds a GCB job invoking it.

Reply:
my bash skills are very basic, could you please add some more comments to this script to explain what it's doing?